File: bin/configure_nginx.py
#! /usr/bin/env python3

import os
import pystache

from pathlib import Path

_template_name = 'nginx.conf.tpl'


def main():
    template_path = Path(os.environ.get('TEMPLATE_DIR'), _template_name)

    with open(template_path, 'r') as template_handle:
        template = template_handle.read()

    template_args = {
        'port': os.environ.get('PORT'),
        'autoenv_http_path': os.environ.get('AUTOENV_HTTP_PATH'),
        'autoenv_fs_path': os.environ.get('AUTOENV_FS_PATH')
    }

    config_path = Path(os.environ.get('NGINX_CONF_PATH'))

    with open(config_path, 'w') as nginx_config_handle:
        nginx_config_handle.write(
            pystache.render(template, template_args)
        )


if __name__ == "__main__":
    main()

File: src/configureNginx.ts
import { template } from 'lodash'
import path from 'path'
import fs from 'fs/promises'

const templatePath = path.join(__dirname, 'templates', 'nginx.conf.tpl')
const configPath = process.env.NGINX_CONF_PATH

const main = async () => {
    const data = await fs.readFile(templatePath, 'utf8')
    const templateFn = template(data)

    const templateArgs = {
        port: process.env.PORT,
        autoenvHttpPath: process.env.AUTOENV_HTTP_PATH,
        autoenvFsPath: process.env.AUTOENV_FS_PATH,
    }

    if (configPath === undefined) {
        throw new Error('Undefined config path')
    }

    fs.writeFile(configPath, templateFn(templateArgs))
}

main()
